commit 11b028a9a01e47fc9b48e4a566803752011902e2
Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
Date:   Wed Nov 13 14:48:38 2024 +0200

    UCM2: Intel: sof-hda-dsp: Fix handling of empty sys_vendor
    
    The mistake in UCM syntax caused in alsaucm start error:
    
    ALSA lib ucm_cond.c:367:(if_eval) unknown If.Condition.Type
    
    Fixes: 13022a97711d ("sof-hda-dsp: Fix the case where sysfs dmi
           sys_vendor attribute is not set")
    
    Closes: https://github.com/alsa-project/alsa-ucm-conf/pull/463
    Signed-off-by: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
    Signed-off-by: Jaroslav Kysela <perex@perex.cz>

diff --git a/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf b/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf
index fea8159..c755ba2 100644
--- a/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf
+++ b/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf
@@ -6,8 +6,8 @@ Define.SOFVendor "$${sys:devices/virtual/dmi/id/sys_vendor}"
 
 If.SOFVendor {
 	Condition {
-		Type Empty
-		String "${var:SOFVendor}"
+		Type String
+		Empty "${var:SOFVendor}"
 	}
 	True.Define.SOFVendor "${sys:devices/virtual/dmi/id/board_vendor}"
 }
commit 6397c663d7086b87ca5cbba323ea3dcd0ecd3200
Author: Jaroslav Kysela <perex@perex.cz>
Date:   Wed Nov 13 16:19:15 2024 +0100

    sof-hda-dsp: Fix the case where sysfs dmi product_name attribute is not set
    
      !!DMI Information
      !!---------------
    
      Manufacturer:
      Product Name:
      Product Version:
      Firmware Version:  R6G07
      System SKU:        Default string
      Board Vendor:      Default string
      Board Name:        Default string
    
    BugLink: https://bugzilla.redhat.com/show_bug.cgi?id=2292583
    Signed-off-by: Jaroslav Kysela <perex@perex.cz>

diff --git a/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf b/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf
index c755ba2..5fd1843 100644
--- a/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf
+++ b/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf
@@ -3,6 +3,7 @@
 # File paths for controlling SOF processing
 
 Define.SOFVendor "$${sys:devices/virtual/dmi/id/sys_vendor}"
+Define.SOFProduct "$${sys:devices/virtual/dmi/id/product_name}"
 
 If.SOFVendor {
 	Condition {
@@ -12,6 +13,14 @@ If.SOFVendor {
 	True.Define.SOFVendor "${sys:devices/virtual/dmi/id/board_vendor}"
 }
 
+If.SOFProduct {
+	Condition {
+		Type String
+		Empty "${var:SOFProduct}"
+	}
+	True.Define.SOFProduct "${sys:devices/virtual/dmi/id/board_name}"
+}
+
 If.SOFIPCVer {
 	Condition {
 		Type ControlExists
@@ -43,7 +52,7 @@ If.SOFPath {
 		HeadphoneIirBlob "${var:BlobPath}/eq_iir/pass.blob"
 		HeadphoneFirBlob "${var:BlobPath}/eq_fir/pass.blob"
 		HeadphoneDrcBlob "${var:BlobPath}/drc/passthrough.blob"
-		ConfPathFromDMI "${var:SOFVendor}/${sys:devices/virtual/dmi/id/product_name}.conf"
+		ConfPathFromDMI "${var:SOFVendor}/${var:SOFProduct}"
 		SOFProductConfig "/blobs/sof/product_configs/${var:ConfPathFromDMI}"
 		SOFConfFullPath "${ConfTopDir}${var:SOFProductConfig}"
 		SOFUserConfig "/blobs/sof/user_configs/${var:ConfPathFromDMI}"
commit 4a01327cc43a18124c0a4a5bcfb840e6ed1efe0a
Author: Jaroslav Kysela <perex@perex.cz>
Date:   Wed Dec 4 10:20:51 2024 +0100

    sof-hda-dsp: Add back missing .conf suffix for product/user specific configs
    
    Fixes: 6397c66 ("sof-hda-dsp: Fix the case where sysfs dmi product_name attribute is not set")
    Closes: https://github.com/alsa-project/alsa-ucm-conf/issues/482
    Signed-off-by: Jaroslav Kysela <perex@perex.cz>

diff --git a/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf b/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf
index 5fd1843..3a4c445 100644
--- a/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf
+++ b/ucm2/Intel/sof-hda-dsp/HiFi-sof.conf
@@ -52,7 +52,7 @@ If.SOFPath {
 		HeadphoneIirBlob "${var:BlobPath}/eq_iir/pass.blob"
 		HeadphoneFirBlob "${var:BlobPath}/eq_fir/pass.blob"
 		HeadphoneDrcBlob "${var:BlobPath}/drc/passthrough.blob"
-		ConfPathFromDMI "${var:SOFVendor}/${var:SOFProduct}"
+		ConfPathFromDMI "${var:SOFVendor}/${var:SOFProduct}.conf"
 		SOFProductConfig "/blobs/sof/product_configs/${var:ConfPathFromDMI}"
 		SOFConfFullPath "${ConfTopDir}${var:SOFProductConfig}"
 		SOFUserConfig "/blobs/sof/user_configs/${var:ConfPathFromDMI}"
